rosbag2_transport/src/rosbag2_transport/player.hpp

```diff
@@ -59,6 +61,17 @@ class Player
   void play_messages_until_queue_empty(const PlayOptions & options);
   void prepare_publishers();
 
+  /**
+   * Determine which QoS to offer for a topic.
+   * The priority of the profile selected is:
+   *   1. The override specified in play_options (if one exists for the topic).
+   *   2. The default RMW QoS profile.
+   *
+   * \param topic_name The full name of the topic, with namespace (ex. /arm/joint_status).
+   * @return The QoS profile to be used for subscribing.
+   */
+  rclcpp::QoS publisher_qos_for_topic(const std::string & topic_name);
```

Comment:
I've just seen that this function is private. You might want to consider replacing this with a free function, taking two arguments (the overrides and topic name).

Reply:
The overrides are specified in the `play_options` which I think is a suitable entrypoint for users. I don't think we'd want to expose the internals here otherwise (Unless a user want's to change QoS policies dynamically...?)

Reply:
I'd agree with Anas that this function is not useful outside the context of the player. It only contains logic that deals with the internal mechanics of the Player. I'll be having it call out to an adaptive function in the case an override is not provided, but that adaptive logic will live in a separate utility.

Reply:
I guess I wasn't clear. I would still keep the (free) function within context of player. But only have it defined in the `player.cpp` and not exposed in the header. Basically for the same reasons you all said, that it doesn't make much sense for users to call this function (which they couldn't anyway since it's private). 

The only class member variable this function is using is `topic_qos_profile_overrides_` which could easily be passed into the free function.

Reply:
Oh, yeah that would be fine